test/fixtures/programmatic/plugin/extend-config-plugin/index.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const pluginConfig = {
+  targetValuePath: ['custom', 'extend', 'value'],
+  overwriteValuePath: ['custom', 'extend', 'overwrite'],
+  afterInitValuePath: ['custom', 'extend', 'afterInit'],
+  refValuePath: ['custom', 'extend', 'ref'],
+};
+
+module.exports = class TestPlugin {
+  constructor(serverless, options, utils) {
+    this.serverless = serverless;
+    this.options = options;
+    this.utils = utils;
+
+    this.hooks = {
+      initialize: () => this.extendAfterInit(),
+    };
+  }
+
+  async asyncInit() {
+    const configExt = {
+      var: 'value',
+    };
+    this.serverless.extendConfiguration(pluginConfig.targetValuePath, configExt);
+    this.serverless.extendConfiguration(pluginConfig.overwriteValuePath, configExt);
+    this.serverless.extendConfiguration(pluginConfig.refValuePath, '${self:custom.extend.value}');
+
+    try {
+      this.serverless.extendConfiguration([], { custom: {} });
+    } catch (error) {
+      // ignore this
+    }
```

Comment:
Let's remove that one. 

It'll make sense if we'd test the error case. but we're catching with without any confirmation on if it thrown and if, with what error

Reply:
How should i test that if the same test needs serverless to fail and succeed at the same time?

Reply:
Each errors are tested with individual `serverless` run. There are examples in same test file

Reply:
I previously had all this. You wanted it to be in one run.
Now please explain exactly how you want it to be done without making the fixture or the tests more complex.
I'm wasting way to much time on this feature.

Reply:
I think we're fine to not have those error scenarios tested. Otherwise the only way to test them is as I mentioned with individual `serverless` invocations

Reply:
In fact they are tested implicitly because they would change the checked outputs what would lead the expects to fail.